server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1186,6 +1219,290 @@ protected void pushToUpdateQueue(final List<NetworkVO> networks) throws Interrup
+            }
+        }
+    }
+
+    private boolean updateRouterHealthCheckData(DomainRouterVO router) {
+        if (!RouterHealthChecksEnabled.valueIn(router.getDataCenterId())) {
+            return false;
+        }
+
+        String controlIP = getRouterControlIP(router);
+        if (StringUtils.isNotBlank(controlIP) && !controlIP.equals("0.0.0.0")) {
+            s_logger.info("Updating data for router health checks for router " + router.getUuid());
+            final SetMonitorServiceCommand command = new SetMonitorServiceCommand();
+            command.setAccessDetail(NetworkElementCommand.ROUTER_IP, getRouterControlIP(router));
+            command.setAccessDetail(NetworkElementCommand.ROUTER_NAME, router.getInstanceName());
+            command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_ENABLED, RouterHealthChecksEnabled.valueIn(router.getDataCenterId()).toString());
+            command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_BASIC_INTERVAL, RouterHealthChecksBasicInterval.value().toString());
+            command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_ADVANCED_INTERVAL, RouterHealthChecksAdvancedInterval.value().toString());
+            command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_EXCLUDED, RouterHealthChecksToExclude.valueIn(router.getDataCenterId()));
+            command.setAdditionalData(getAdditionalDataForRouterHealthChecks(router));
+            command.setReconfigureAfterUpdate(true);
+
+            Answer origAnswer = null;
+            try {
+                origAnswer = _agentMgr.easySend(router.getHostId(), command);
+            } catch (final Exception e) {
+                s_logger.warn("Error while collecting alerts from router: " + router.getInstanceName(), e);
+                return false;
+            }
+
+            if (origAnswer == null) {
+                s_logger.warn("Unable to update health checks data to router " + router.getHostName());
+                return false;
+            }
+
+            GroupAnswer answer = null;
+            if (origAnswer instanceof GroupAnswer) {
+                answer = (GroupAnswer) origAnswer;
+            } else {
+                s_logger.warn("Unable to update health checks data to router " + router.getHostName() + " Received answer " + origAnswer.getDetails());
+                return false;
+            }
+
+            if (!answer.getResult()) {
+                s_logger.warn("Unable to update health checks data to router " + router.getHostName() + ", details : " + answer.getDetails());
+            }
+
+            return answer.getResult();
+        }
+        s_logger.debug("Skipping update data on router " + router.getUuid() + " because controlIp is not correct.");
+        return false;
+    }
+
+    private Map<String, String> getAdditionalDataForRouterHealthChecks(final DomainRouterVO router) {
```

Comment:
This is also a big method, can it be split please?

Reply:
Done

Reply:
still 40 line ;)